File: Examples/Conversations/Obj-C/ConversationsExample/CommentsViewController.m
//
//  CommentsViewController.m
//  ConversationsExample
//
//  Copyright © 2017 Bazaarvoice. All rights reserved.
//

#import "CommentsViewController.h"
#import "BVComment.h"
#import "MyCommentTableViewCell.h"

@interface CommentsViewController () <UITableViewDataSource>

@property(weak, nonatomic) IBOutlet UITableView *commentsTableView;

@property NSArray<BVComment *> *comments;

@end

@implementation CommentsViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  self.comments = [NSArray array];

  self.commentsTableView.dataSource = self;
  self.commentsTableView.estimatedRowHeight = 44;
  self.commentsTableView.rowHeight = UITableViewAutomaticDimension;
  [self.commentsTableView
                 registerNib:[UINib nibWithNibName:@"MyCommentTableViewCell"
                                            bundle:nil]
      forCellReuseIdentifier:@"MyCommentTableViewCell"];

  BVCommentsRequest *request =
      [[BVCommentsRequest alloc] initWithReviewId:@"192548" limit:99 offset:0];

  [request load:^(BVCommentsResponse *_Nonnull response) {
    // success
    self.comments = response.results;
    [self.commentsTableView reloadData];

  }
      failure:^(NSArray<NSError *> *_Nonnull errors) {
        // error
        NSLog(@"ERROR Loading Comments: %@",
              errors.firstObject.localizedDescription);
      }];
}

#pragma mark UITableViewDatasource

- (NSString *)tableView:(UITableView *)tableView
    titleForHeaderInSection:(NSInteger)section {
  return @"Comments";
}

- (NSInteger)tableView:(UITableView *)tableView
    numberOfRowsInSection:(NSInteger)section {
  return [self.comments count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {

  MyCommentTableViewCell *cell =
      [tableView dequeueReusableCellWithIdentifier:@"MyCommentTableViewCell"];

  cell.comment = [self.comments objectAtIndex:indexPath.row];

  return cell;
}

@end

File: Examples/Conversations/Swift/ConversationsExample/CommentsViewController.swift
//
//  CommentsViewController.swift
//  
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

import UIKit
import BVSDK

class CommentsViewController: UIViewController, UITableViewDataSource {
  
  
  @IBOutlet weak var commentsTaleView: UITableView!
  var comments : [BVComment] = []
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    commentsTaleView.dataSource = self
    commentsTaleView.estimatedRowHeight = 44
    commentsTaleView.rowHeight = UITableViewAutomaticDimension
    commentsTaleView.register(UINib(nibName: "MyCommentTableViewCell", bundle: nil), forCellReuseIdentifier: "MyCommentTableViewCell")
    
    let limit : UInt16  = 99
    let offset : UInt16 = 0
    let request = BVCommentsRequest(productId: Constants.TEST_REVIEW_PRODUCT_ID_FOR_COMMENTS, andReviewId: Constants.TEST_REVIEW_ID_FOR_COMMENTS, limit: limit, offset: offset)
    
    request.load({ (response) in
      
      self.comments = response.results
      self.commentsTaleView.reloadData()
      
    }) { (error) in
      
      print("ERROR fetching comments: \(error.first!.localizedDescription)")
      
    }
    
  }
  
  override func didReceiveMemoryWarning() {
    super.didReceiveMemoryWarning()
    // Dispose of any resources that can be recreated.
  }
  
  // MARK: UITableViewDatasource
  
  func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
    return "Available Comments"
  }
  
  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return comments.count
  }
  
  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
    let tableCell = tableView.dequeueReusableCell(withIdentifier: "MyCommentTableViewCell") as! MyCommentTableViewCell
    
    tableCell.comment = comments[indexPath.row]
    
    return tableCell
  }
  
  
  
}
